stdlib/universe/histogram_quantile.go

```diff
@@ -251,51 +251,97 @@ func (t histogramQuantileTransformation) Process(id execute.DatasetID, tbl flux.
 		})
 	}
 
-	q, ok, err := t.computeQuantile(cdf)
+	result, err := t.computeQuantile(cdf)
 	if err != nil {
 		return err
 	}
-	if !ok {
+	if result.action == drop {
 		return nil
 	}
 	if err := execute.AppendKeyValues(tbl.Key(), builder); err != nil {
 		return err
 	}
-	if err := builder.AppendFloat(valueIdx, q); err != nil {
-		return err
+	if result.action == appendValue {
+		if err := builder.AppendFloat(valueIdx, result.v); err != nil {
+			return err
+		}
+	} else {
+		// action is appendNil
+		if err := builder.AppendNil(valueIdx); err != nil {
+			return err
+		}
+
 	}
 	return nil
 }
 
-func (t *histogramQuantileTransformation) computeQuantile(cdf []bucket) (float64, bool, error) {
+type quantileAction int
+
+const (
+	appendValue quantileAction = iota
+	appendNil
+	drop
+)
+
+type quantileResult struct {
+	action quantileAction
+	v      float64
+}
+
+// isMonotonic will check if the buckets are monotonic and
+// return true if so.
+//
+// If force is set, it will force them to be monotonic
+// by assuming no increase from the previous bucket.
+// When force is set, this function will always return true.
+func isMonotonic(force bool, cdf []bucket) bool {
+	prevCount := 0.0
+	for i := range cdf {
+		if cdf[i].count < prevCount {
+			if force {
+				cdf[i].count = prevCount
+			} else {
+				return false
+			}
+		} else {
+			prevCount = cdf[i].count
+		}
+	}
+	return true
+}
+
+func (t *histogramQuantileTransformation) computeQuantile(cdf []bucket) (quantileResult, error) {
```

Comment:
The issue here was that the Flux stdlib function `histogram_quantile` gets a wrong answer for some input data.

This function accepts a cumulative distribution function (a cumulative histogram produced from the input table data) and produces the requested quantile.

When the cdf contains all zeroes, this function would return the bound of the last histogram bucket, which is incorrect. The right thing to do for that case is to return a null value, since we can't compute a quantile if we didn't actually receive any observations.